[docs] Add guide on deployment to Google cloud (#3387)
mui/mui-toolpad: docs/data/toolpad/studio/how-to-guides/gcp-deploy.md

## Prerequisites

- A [GCP](https://console.cloud.google.com/) account
- A [GitHub](https://github.com) account
- A GitHub repository containing your Toolpad Studio app. Check out [pushing your Toolpad Studio app to GitHub](/toolpad/studio/how-to-guides/render-deploy/#pushing-your-toolpad-studio-app-to-github) for this step.

## Setting a new app on GCP

1. Login in to Google Cloud console and from the header bar choose the project where you want to setup the web service.

{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-gcp/choose-project.png", "alt": "Choose project", "caption": "Choosing the project to setup the web service", "indent": 1 }}

2. From the navigation menu icon, click on **Cloud Run**.

{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-gcp/navigate-to-cloud-run.png", "alt": "Navigate to cloud run", "caption": "Navigate to cloud run", "indent": 1}}

> 3. Click on **Create Service** to setup your web app.

Reviewer: ```suggestion
3. Click on **Create Service** to set up your web app.
```

{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-gcp/create-service.png", "alt": "Create Cloud Run service", "caption": "Create Cloud Run service", "indent": 1}}

> 4. In the Create service UI, choose **Continuously deploy from a repository** and give a name to your service from **Service name** input. Further click on **Setup with cloud build** button.

Reviewer: ```suggestion
4. In the Create service UI, choose **Continuously deploy from a repository** and give a name to your service from the **Service name** input. Further click on **Setup with cloud build** button.
```

{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-gcp/setup-run-service.png", "alt": "Configuring Cloud Run service", "caption": "Configuring Cloud Run service", "indent": 1 }}

> 5. This opens a drawer menu, from the drop-down, choose the repository that contains your Toolpad Studio app and click Next.

Reviewer: ```suggestion
5. This opens a drawer menu. From the drop-down, choose the repository that contains your Toolpad Studio app and click Next.
```

{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-gcp/build-type.png", "alt": "Choosing branch and build type", "caption": "Choosing branch and build type", "indent": 1 }}

7. With the drawer closed, on the Cloud Run setup UI, choose **Allow unauntheticated invocations** to create a public app that needs no authentication and scroll down.

{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-gcp/back-to-cloud-run.png", "alt": "Select allow unauntheticated invocations", "caption": "Select allow unauntheticated invocations", "indent": 1}}

8. The last step is to click on 'Container(s), Volumes, Networking, Security' carrot. The first input **Container port** needs to be updated to 3000. Click the **Create** button to start the service.

{{"component": "modules/components/DocsImage.tsx", "src": "/static/toolpad/docs/studio/deploy-gcp/set-container-port.png", "alt": "Set contianer port", "caption": "Set contianer port", "indent": 1}}

> 9. The next screen shows that the deployment is in progress and the build is being prepared, it can take 3-5 minutes. You can click on the **logs** link (pending state) to see the live build logs in Cloud Build UI.

Reviewer: ```suggestion
9. The next screen shows that the deployment is in progress and the build is being prepared, which can take 3-5 minutes. You can click on the **logs** link (pending state) to see the live build logs in Cloud Build UI.
```
